torch/_dynamo/variables/user_defined.py

```diff
@@ -1107,20 +1107,25 @@ def var_getattr(self, tx: "InstructionTranslator", name):
         ):
             # Attribute has a __get__ method. Create a user defined object vt
             # for the subobj, and then trace the __get__ method.
-            descriptor_var = UserDefinedObjectVariable(subobj, source=source)
-
-            get_source = self.source
-            if self.source:
-                get_source = AttrSource(self.source, "__get__")
+            descriptor_source = None
+            descriptor_get_source = None
+            if self.cls_source:
+                # To access the method descriptor from the udf object w/o using
+                # inspect.getattr_static, we can look into the class __dict__
+                descriptor_source = GetItemSource(
+                    AttrSource(self.cls_source, "__dict__"), name
+                )
+                descriptor_get_source = AttrSource(descriptor_source, "__get__")
+            descriptor_var = UserDefinedObjectVariable(subobj, source=descriptor_source)
```

Comment:
Can we use builder here?

Reply:
For sourceless, I cant rely on the SourcelessBuilder because it is a UserDefinedObject. But, let me try it for sourceful.

Reply:
Done